docs/source/getting-started/intro.rst

```diff
@@ -0,0 +1,312 @@
+with data sets and statistics, while still leveraging the power of
+parallel processing when necessary. Motivated by the power of
+`Jupyter notebooks`_, we adopt a "lab" framework for nbodykit, where all of the
+necessary data containers and algorithms can be imported from a single module:
+
+.. code-block:: python
+
+  from nbodykit.lab import *
+
+  [insert cool science here]
+
+See the documentation for :mod:`nbodykit.lab` for a full list of the
+imported members in this module.
+
+With all of the necessary tools now in hand, the user can easily load
+a data set, compute statistics of that data via one of the
+built-in algorithms, and save the results in just a few lines. The end
+result is a reproducible scientific result, generated from clear
+and concise code that flows from step to step.
+
+Parallel computation with MPI
+-----------------------------
+
+The nbodykit package is fully parallelized using the Python
+bindings of the Message Passage Interface (MPI) available in ``mpi4py``. While
+we aim to hide most of the complexities of MPI from the top-level user
+interface, it is helpful to know some basic aspects of the MPI framework
+for understanding how nbodykit works to compute its results. If you are
+unfamiliar with MPI, a good place to start is the `documentation for
+mpi4py <http://mpi4py.readthedocs.io/en/stable/intro.html>`_. Briefly,
+MPI allows nbodykit to use a specified number of CPUs, which work independently
+to achieve a common goal and pass messages back and forth to coordinate their
+work.
+
+When writing code with nbodykit, it's important to keep in mind that memory is
+not shared across different CPUS when using MPI. This is particularly relevant
+when loading data in parallel using nbodykit, as the data is spread out
+evenly across all of the available CPUs. This allows nbodykit to load
+large data sets that would otherwise not fit into the memory of a single CPU,
+given the prohibitively large size. However, a single CPU does not have access
+to the full dataset, but merely the portion stored in its memory
+(usually :math:`1/N` of the full data set, where :math:`N` is the number of CPUs).
+
+Users can execute Python scripts using nbodykit and MPI using the standard
+``mpiexec`` executable. For example, we can run this nbodykit
+`example script <https://raw.githubusercontent.com/bccp/nbodykit/master/nersc/example.py>`_
+with 4 MPI workers using:
+
+.. code:: bash
+
+    # first download the example script
+    $ wget https://raw.githubusercontent.com/bccp/nbodykit/master/nersc/example.py
+
+    # run with 4 MPI workers
+    $ mpiexec -n 4 python example.py
+
+The MPI calling sequence differs for supercomputing environments and depends
+on the task scheduler being used. For example, when using the
+`Slurm manager <https://slurm.schedmd.com>`_, the equivalent command to
+execute the example script is:
+
+.. code:: bash
+
+    $ srun -n 4 python example.py
```

Comment:
On a managed computing facility, it is also usually necessary to include directives that reserves the computing resource used for running the script. Usually you can refer to the 'Submitting a Job' section of the user guide of the facility. (e.g. add a link to NERSC and NCSA; though be aware these user guides are not always accurate., and it is always better to check with someone who uses these facilities first.)